api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -47,18 +52,69 @@
   private static final String SAM_STATUS_NAME = "Sam";
   private static final String RAWLS_STATUS_NAME = "Rawls";
   private static final String GOOGLE_BUCKETS_STATUS_NAME = "GoogleBuckets";
+  private static final int MAX_ATTEMPTS = 3;
 
   @Autowired
   public FireCloudServiceImpl(Provider<WorkbenchConfig> configProvider,
       Provider<ProfileApi> profileApiProvider,
       Provider<BillingApi> billingApiProvider,
       Provider<GroupsApi> groupsApiProvider,
-      Provider<WorkspacesApi> workspacesApiProvider) {
+      Provider<WorkspacesApi> workspacesApiProvider,
+      Sleeper sleeper) {
     this.configProvider = configProvider;
     this.profileApiProvider = profileApiProvider;
     this.billingApiProvider = billingApiProvider;
     this.groupsApiProvider = groupsApiProvider;
     this.workspacesApiProvider = workspacesApiProvider;
+    this.sleeper = sleeper;
+  }
+
+  private static boolean isServiceUnavailableException(Exception e) {
+    if (e instanceof ApiException) {
+      int code = ((ApiException) e).getCode();
+      return (code > 500 && code <= 600);
+    }
+    return false;
+  }
+
+  private <T> T callFirecloudWithRetries(Callable<T> callable) throws ApiException {
+    int numAttempts = 0;
+    // Retry on 503 exceptions.
+    while (true) {
+      try {
+        return callable.call();
+      } catch (Exception e) {
+        numAttempts++;
+        if (isServiceUnavailableException(e)) {
+          if (numAttempts < MAX_ATTEMPTS) {
+            log.log(Level.WARNING,
+                String.format("Firecloud unavailable, attempt %s; retrying...", numAttempts), e);
+            try {
+              // Sleep with some backoff.
+              sleeper.sleep(2000 * numAttempts );
```

Comment:
This should jitter and should probably use exponential backoff.

opt: These libraries should be available in our build:

https://developers.google.com/api-client-library/java/google-http-java-client/backoff
https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/util/ExponentialBackOff

Reply:
Those are used at the HTTP client level... we're trying to do retries using Swagger or Google client libs in these cases.

I'm going to look into using Spring retry here:

https://docs.spring.io/spring-batch/trunk/reference/html/retry.html